### PR TITLE
Add metrics per keys for next scheduled attestation and proposal

### PR DESCRIPTION
**Pull request by killerwhile**
**What type of PR is this?**

> Uncomment one line below and remove others.
>
> Bug fix
Feature
> Documentation
> Other

**What does this PR do? Why is it needed?**

This PR adds metrics for the next scheduled proposal and attestation slot, per keys.
This information is available today as a log line, it would be more convenient to have it as a metric, because it would allows programmatic automation, such as when to schedule the next validator restart/maintenance.

**Which issues(s) does this PR fix?**
-

**Other notes for review**

That's my first PR 🎉

I did struggle a bit between keys and truncated keys, I'm building a map for the lookup, let me know if you have better ideas to address that. Thanks!

### CODE REVIEW

File: validator/client/validator.go
Comment (terencechain) [inline validator/client/validator.go:655]: Aren't key and value here basically the same thing?

File: validator/client/validator.go
Comment (terencechain) [inline validator/client/validator.go:670]: I'm concern about this potential o(n^2) loop. Let's optimize it to o(n) if we can

**Review by terencechain — CHANGES_REQUESTED** [review]

**Comment by terencechain** [comment]
Thanks for the PR! Just a few minor feedbacks

**Review by killerwhile — COMMENTED** [review]
Thanks for the review. Based on your advice, I found a better place where to update the new counters, avoiding the map and the loop de loop at the same time.
Please let me know WDYT about this version. Thanks!

**Review by terencechain — APPROVED** [review]
LGTM. Thanks!

**Comment by rauljordan** [comment]
Hi @killerwhile we do not have permissions to update your branch with the base branch, could you enable this in the settings of your fork? Otherwise the PR cannot merge unless it is fully up to date and we will need to keep reminding you to hit the update button if the base branch changes (it changes a lot throughout the day). Thanks for the contribution!

File: validator/client/validator.go
Comment (prestonvanloon) [inline validator/client/validator.go:665]: This needs to be wrapped with if v.emitAccountMetrics {}
```suggestion
if v.emitAccountMetrics {			 
  ValidatorNextProposalSlotGaugeVec.WithLabelValues(validatorNotTruncatedKey).Set(float64(proposerSlot))
}
```

File: validator/client/validator.go
Comment (prestonvanloon) [inline validator/client/validator.go:655]: This needs to be wrapepd with if v.emitAccountMetrics

```suggestion
			if v.emitAccountMetrics {
			ValidatorNextAttestationSlotGaugeVec.WithLabelValues(validatorNotTruncatedKey).Set(float64(duty.AttesterSlot))
			}
```

Please excuse my formatting, github doesn't do a great job with formatting in suggestions. Please run gofmt after accepting these changes.

**Review by prestonvanloon — CHANGES_REQUESTED** [review]
Please wrap these metric updates with if v.emitAccountMetrics. 
The metrics page and influx db can grow out of control for stakers a high number of public keys, so we only want to emit such cardinal metrics if the user hasn't opted out.

File: validator/client/validator.go
Comment (killerwhile) [reply validator/client/validator.go]: Fixed as requested.

**Comment by killerwhile** [comment]
@rauljordan sorry for that, I understand the problem. I'm actually not able to find this option (it's probably time to go to bed for me :)), if you have a small hint on what I missed I'd appreciate. Otherwise I just rebased with my latest changes, if timing is good.

**Comment by rauljordan** [comment]
Hi @killerwhile , hmm actually I was also unable to find it, weird. Typically, we are able to update contributor's branches. We'll review asap after you update the branch again and try to merge soon

**Comment by rauljordan** [comment]
PTAL @prestonvanloon

**Comment by killerwhile** [comment]
Friendly reminder @prestonvanloon. Thanks!

**Comment by rauljordan** [comment]
Hi @killerwhile it will be quite difficult for us to merge this given how much activity happens in the codebase and every update will require us to contact you to coordinate. We should look into options on how we can automatically update your branch. For most contributors, this option is available to us automatically, so it leads us to think this has to do with your settings

**Comment by rauljordan** [comment]
Screen Shot 2021-03-08 at 9 12 27 PM
For example, we cannot hit "update branch", whereas for other contributor's PRs and forks, we are able to do so

**Comment by killerwhile** [comment]
Thanks for your patience and deeply apologize for the noise.
I created #8583 instead, from my personal GH account, and this time I found the "Allow maintainers' edit" checkbox.
The other PR was from an ORG and apparently I'd have needed to create a team with you in to allows updates on PR.
Anyway, I'm closing this one in favor of #8583.

**Pull request closed without merging**
